src/core/Form/TextInput/TextInput.tsx

```diff
@@ -64,8 +71,31 @@ const StyledTextInput = styled(
  */
 export class TextInput extends Component<TextInputProps> {
   render() {
-    const { ...passProps } = withSuomifiDefaultProps(this.props);
+    const {
+      children,
+      icon,
+      iconProps,
+      className,
+      ...passProps
+    } = withSuomifiDefaultProps(this.props);
+
+    const prioritizedIcon = icon || iconProps?.icon;
```

Comment:
prioritizedIcon sound misleading to me. If not possible to use icon, I'd prefer resolvedIcon or something like that.

Reply:
Your suggestion sounds like a better name. I'll go with that.